src/libserialize/serialize.rs

```diff
@@ -895,3 +896,6 @@ impl<T: UseSpecializedDecodable> Decodable for T {
 impl<'a, T: ?Sized + Encodable> UseSpecializedEncodable for &'a T {}
 impl<T: ?Sized + Encodable> UseSpecializedEncodable for Box<T> {}
 impl<T: Decodable> UseSpecializedDecodable for Box<T> {}
+
+impl<'tcx> UseSpecializedEncodable for PhantomData<&'tcx ()> {}
+impl<'tcx> UseSpecializedDecodable for PhantomData<&'tcx ()> {}
```

Comment:
Huh, I wouldn't think `PhantomData` would be relevant.
What about just fully implementing encoding/decoding for PhantomData, regardless of parameter?